docs/pages/dsDisplay/ds-display_L3_Test-Procedure.md

```diff
@@ -0,0 +1,305 @@
+            password: ' '
+        - ssh_hal_test:
+            type: "ssh"
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+      outbound:
+        download_url: "tftp://tftp-server.com/rack1/slot1/"    # Download location for the CPE device
+        upload_url: "sftp://server-address/home/workspace/tftp/rack1/slot1/" # Upload location
+        upload_url_base_dir: "sftp://server-address/home/workspace/tftp/rack1/slot1"
+        httpProxy:   # Local proxy if required
+        workspaceDirectory: './logs/workspace'   # Local working directory
+
+```
+
+#### Device Configuration File
+
+  Example Device configuration File: `ut/host/tests/configs/deviceConfig.yml`
+
+For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+Update the target directory where `HAL` binaries will be copied into the device. Also, map the profile to the source/sink settings `YAML` file path.
+
+Ensure the platform should match with the `DUT` platform in [Rack Configuration](#rack-configuration-file)
+
+```yaml
+deviceConfig:
+  cpe1:
+    platform: "stb"    # Must match the platform in example_rack_config.yml
+    model: "uk"
+    target_directory: "/tmp"  # Path where HAL binaries are copied in device
+    test:
+      profile: "../../../../profiles/source/Source_4K_Display.yaml"
+      player:
+        tool: "gstreamer"
+        prerequisites:
+          - export xxxx    # Pre-commands required to play the stream
+
+```
+
+#### Test Setup Configuration File
+
+Update the artifact paths from which the binaries should be copied to the device.
+
+Set the execution paths.
+
+- Example configuration file: [dsDisplay_L3_testSetup.yml](../../../host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_testSetup.yml).
+
+#### Test Suite Configuration
+
+Update the execute command according to the device path where `HAL` binaries are copied and Update the test suite for each level test case
+
+- Example configuration file: [dsDisplay_test_suite.yml](../../../host/tests/dsClasses/dsDisplay_test_suite.yml).
+
+## Run Test Cases
+
+Once the environment is set up, you can execute the test cases with the following command
+
+```bash
+python <TestCaseName.py> --config </PATH>/ut/host/tests/configs/example_rack_config.yml --deviceConfig </PATH>/ut/host/tests/configs/deviceConfig.yml
+```
+
+## Test Case Procedure
+
+### dsDisplay_test01_VerifyDisplayConnectCallBackTest.py
+
+**Overview:**
+
+This test verifies display event callbacks by connecting and disconnecting a display device on each supported video port, ensuring that appropriate callbacks are triggered for display connection, disconnection, Rx Sense ON, Rx Sense OFF, and HDCP protocol version change events.
+
+**Platform Supported:**
+
+- Source
+- Sink
```

Comment:
We don't support the Rx Sense events and HDCP for sink devices.  Please confirm what events are used on source and sink platforms.

Reply:
DONE